src/browser/extension/inject/index.js

```diff
@@ -5,7 +5,7 @@ window.devToolsExtensionID = 'lmhkpmbekcpmknklioeibfkpmmfibljd';
 
 chrome.runtime.sendMessage(window.devToolsExtensionID, { type: 'GET_OPTIONS' }, function(response) {
   if (!response.options.inject) {
-    const urls = response.options.urls.split('\n').join('|');
+    const urls = response.options.urls.split('\n').filter(Boolean).join('|');
```

Comment:
This thing is a bit weird, but it’s a quick solution.

So the thing is that previously we saved and synchronized the trimmed content of the `blacklist`, `whitelist` and `url` textareas. When a user entered

```
AN_ACTION
ANOTHER_ACTION
(a new line)
```

into the `blacklist` field, we trimmed it and received `"AN_ACTION\nANOTHER_ACTION"` string. Then we replaced `\n` with `|` and used that string as a regexp. The trimming was necessary because if we wouldn’t do it, we’d receive a `AN_ACTION|ANOTHER_ACTION|` regexp that matched everything. This worked great.

When I moved `Options` into a separate component and made all the inputs controlled, turned out re-rendering an input with the trimmed text effectively prevents a user from putting a new line. The quickest solution for this was to save the non-trimmed version of the text and instead filter out all the empty lines where a regexp is created. `.filter(Boolean) does exactly this: it filters out all falsy values including empty strings.

There’s probably a better solution for this, but I haven’t investigated.